ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,141 @@
+import { Checkbox, FormControlLabel, ListItem } from "@mui/material";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import { CriteriaConfig, CriteriaPlugin, registerCriteriaPlugin } from "cohort";
+import produce from "immer";
+import React from "react";
+import * as tanagra from "tanagra-api";
+import { useAppDispatch, useUnderlay } from "../hooks";
+
+type Selection = {
+  entity: string;
+  id: number;
+  name: string;
+};
+
+
+interface Config extends CriteriaConfig {
+  entity: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    console.log(this.data)
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter(entityVar: string, fromOccurrence: boolean) {
+    return null;
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const dispatch = useAppDispatch();
+  const underlay = useUnderlay();
+  const attribute = props.data.entity;
+  const attributeName = verifyName(attribute);
+
+  function verifyName(demographic: string) {
+    return demographic.concat("_concept_id");
+  }
+
+  function hintDisplayName(hint: tanagra.EnumHintValue) {
+    return hint.displayName || "Unknown Value";
+  }
+
+  const enumHintValues = underlay?.entities
+    .filter((g) => g.name === underlay.primaryEntity)[0]
+    .attributes?.filter((g) => g.name === attributeName)[0].attributeFilterHint
+    ?.enumHint?.enumHintValues;
+  const index =
```

Comment:
This approach works but is difficult to understand. Instead, create a function that finds the index for a single id when needed:

```
const selectionIndex = (hint: tanagra.EnumHintValue) =>
  props.data.selected.findIndex((row) => row.id === hint.attributeValue?.int64Val))
```